rosidl_generator_c/resource/msg__functions.c.em

```diff
@@ -98,11 +102,16 @@ for field in spec.fields:
             elif field.default_value is not None:
                 # set default value of primitive type
                 lines.append('msg->%s = %s;' % (field.name, value_to_c(field.type, field.default_value)))
-
         else:
             # initialize the sub message
-            lines.append('%s__%s__%s__init(&msg->%s);' % (field.type.pkg_name, 'msg', field.type.type, field.name))
-
+            lines.append('if (!%s__%s__%s__init(&msg->%s)) {' % (field.type.pkg_name, 'msg', field.type.type, field.name))
+            lines.append('  return false;')
+            lines.append('}')
+            cleanup_string = ('%s__%s__%s__destroy(&msg->%s);' % (field.type.pkg_name, 'msg', field.type.type, field.name))
+            if field.type.is_primitive_type():
```

Comment:
Isn't this in the `else` block which already ensures that the field is not a primitive type? So this condition should never be true.